File: backend/src/main/java/app/miniappspring/service/DiscountService.java
package app.miniappspring.service;

import app.miniappspring.dto.discount.DiscountCreateDto;
import app.miniappspring.dto.discount.DiscountDto;
import app.miniappspring.dto.product.ProductCardDto;

import java.util.List;

public interface DiscountService {
   DiscountDto createDiscount(DiscountCreateDto discountCreateDto);
   DiscountDto checkDiscountAtProduct(Long productId);

   List<DiscountDto> getDiscountList();

   List<ProductCardDto> getProductsWithDiscountByTown(String town);
}

File: frontend/react-app/src/service/discount/DiscountService.ts
import {ICreateCoupon} from "../../model/coupon/ICreateCoupon";
import {DiscountController} from "../../controller/DiscountController";

export class DiscountService {
    public async checkDiscountAtProduct(productId: string) {
        return await DiscountController.checkDiscountAtProduct(productId);
    }

    public async createDiscount(coupon:ICreateCoupon) {
        return await DiscountController.createDiscount(coupon);
    }
}